src/scenarios/FileUpload.scenario.tsx

```diff
@@ -0,0 +1,112 @@
+import CloudUploadIcon from '@mui/icons-material/CloudUpload'
+import {
+  Box,
+  Button,
+  LinearProgress,
+  Alert,
+  type AlertColor
+} from '@mui/material'
+import React, { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import Layout from '../components/Layout'
+import PageSetup from '../components/PageSetup'
+
+const FileUpload = (): JSX.Element => {
+  const { t } = useTranslation()
+  const [file, setFile] = useState<File | null>(null)
+  const [uploading, setUploading] = useState(false)
+  const [uploadMessage, setUploadMessage] = useState<{
+    text: string
+    severity: AlertColor
+  }>({
+    text: '',
+    severity: 'info'
+  })
+
+  const handleFileChange = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ): void => {
+    if (event.target.files != null && event.target.files.length > 0) {
+      setFile(event.target.files[0])
+      setUploadMessage({ text: '', severity: 'info' })
+    }
+  }
+
+  const simulateFileUpload = (): void => {
+    const min = 1000
+    const max = 10000
+    setTimeout(
+      () => {
+        setUploading(false)
+        setUploadMessage({
+          text: t('scenarios.file-upload.upload-success'),
+          severity: 'success'
+        })
+      },
+      Math.floor(Math.random() * (max - min + 1) + min)
+    )
+  }
+
+  const handleUpload = (): void => {
+    if (file != null) {
+      setUploading(true)
+      simulateFileUpload()
+    } else {
+      setUploadMessage({
+        text: t('scenarios.file-upload.please-select-file'),
+        severity: 'error'
+      })
+    }
+  }
+
+  return (
+    <Layout>
+      <PageSetup
+        title={t('scenarios.file-upload.title')}
+        description={t('scenarios.file-upload.description')}
+        information={t('scenarios.file-upload.information')}
+      />
+      <Box>
+        <input
+          type='file'
+          onChange={handleFileChange}
+          id='icon-button-file'
+          style={{ display: 'none' }}
+          data-testid='select-file'
+        />
+        <label htmlFor='icon-button-file'>
+          <Button
+            variant='outlined'
+            component='span'
+            sx={{ mr: 4 }}
+            startIcon={<CloudUploadIcon />}
+          >
+            {t('scenarios.file-upload.select-file')}
+          </Button>
+        </label>
+        <Button
+          variant='contained'
+          color='primary'
+          onClick={handleUpload}
+          disabled={uploading}
+          data-testid='upload-button'
+        >
+          {t('scenarios.file-upload.upload-button')}
+        </Button>
+        {uploading && <LinearProgress sx={{ mt: 2 }} />}
```

Comment:
How about showing the selected file ?
        <Typography variant='body2'>
            {file?.name}
        </Typography>

Reply:
Added the filename to the translated string now.

![image](https://github.com/nearform/testing-playground/assets/2647926/abb23349-152d-4fe1-aba3-2b31490e52e3)

Reply:
Nice !!!
But I was thinking of showing the file name when it is selected. Before the upload.

<img width="361" alt="Screenshot 2023-11-07 at 09 17 36" src="https://github.com/nearform/testing-playground/assets/110396367/7ea65bb5-dba8-450f-a457-d2be03f6e766">

Reply:
Ahh ok, will fix that now.

Reply:
I've updated the behaviour now as well, so the upload button only appears when a file has been selected.

![image](https://github.com/nearform/testing-playground/assets/2647926/3b14a52a-e12e-43ba-88b2-d50ef50156ed)